ScanbotSDKExamples/NET/ReadyToUseUI.Droid/MainActivity.cs

```diff
@@ -32,115 +29,172 @@
 
         protected override void OnCreate(Bundle savedInstanceState)
         {
             base.OnCreate(savedInstanceState);
 
             // Set our view from the "main" layout resource
-            SetContentView(Resource.Layout.Main);
+            SetContentView(Resource.Layout.activity_main);
 
-            sdkInstance = new IO.Scanbot.Sdk.ScanbotSDK(this);
+            scanbotSDK = new IO.Scanbot.Sdk.ScanbotSDK(this);
+            pageStorage = scanbotSDK.CreatePageFileStorage();
 
             var container = (LinearLayout)FindViewById(Resource.Id.container);
-
             var title = container.FindViewById<TextView>(Resource.Id.textView);
             title.Text = Texts.scanbot_sdk_demo;
 
-            progress = FindViewById<ProgressBar>(Resource.Id.progressBar);
-
             var scanner = (LinearLayout)container.FindViewById(Resource.Id.document_scanner);
             var scannerTitle = (TextView)scanner.FindViewById(Resource.Id.textView);
-            scannerTitle.Text = DocumentScanner.Instance.Title;
-            AddItemsTo(scanner, DocumentScanner.Instance.Items);
-
-            var barcodes = (LinearLayout)container.FindViewById(Resource.Id.barcode_detectors);
-            var barcodeTitle = (TextView)barcodes.FindViewById(Resource.Id.textView);
-            barcodeTitle.Text = BarcodeDetectors.Instance.Title;
-            AddItemsTo(barcodes, BarcodeDetectors.Instance.Items);
+            scannerTitle.Text = "DOCUMENT SCANNER".ToUpper();
+            scanner.AddChildren(buttons, new[]
+            {
+                new ListItemButton(this, "Scan Document", ScanDocument),
+                new ListItemButton(this, "Scan Document with Finder", ScanDocumentWithFinder),
+                new ListItemButton(this, "Import Image", ImportImage),
+                new ListItemButton(this, "View Images", ViewImages)
+            }); 
 
             var detectors = (LinearLayout)container.FindViewById(Resource.Id.data_detectors);
             var detectorsTitle = (TextView)detectors.FindViewById(Resource.Id.textView);
-            detectorsTitle.Text = DataDetectors.Instance.Title;
-            AddItemsTo(detectors, DataDetectors.Instance.Items);
-
-            LicenseIndicator.Text = Texts.no_license_found_the_app_will_terminate_after_one_minute;
-        }
-
-        void AddItemsTo(LinearLayout container, List<ListItem> items)
-        {
-            foreach (ListItem item in items)
+            detectorsTitle.Text = "DATA DETECTORS".ToUpper();
+            detectors.AddChildren(buttons, new[]  
             {
-                var child = new FragmentButton(this)
-                {
-                    Data = item,
-                    Text = item.Title,
-                    LayoutParameters = ViewUtils.GetParameters(this)
-                };
-                container.AddView(child);
-                buttons.Add(child);
-            }
-        }
+                new ListItemButton(this, "Scan QR-/Barcode", ScanBarcode),
```

Comment:
@yurii-scanbot I think we should keep barcode related scanners under its own heading (as it was previously) as it is structured this way in the other examples and the documentation (and it also reflects our billing structure for the packages we offer).